File: src/main/java/me/xjcyan1de/cyanbot/gui/commands/CommandChat.java
package me.xjcyan1de.cyanbot.gui.commands;

import com.intellij.uiDesigner.core.GridConstraints;
import com.intellij.uiDesigner.core.GridLayoutManager;
import me.xjcyan1de.cyanbot.Bot;
import me.xjcyan1de.cyanbot.Main;

import javax.swing.*;
import java.awt.*;

public class CommandChat extends Command {

    public CommandChat() {
        super("Сообщение в чат");
    }

    @Override
    public void initPanel(JPanel commandPanel) {
        JPanel chatPanel = new JPanel();
        chatPanel.setLayout(new GridLayoutManager(2, 1, new Insets(0, 0, 0, 0), -1, -1));
        chatPanel.setBorder(BorderFactory.createTitledBorder("Чат"));

        JTextField chatText = new JTextField();
        JButton chatButton = new JButton();
        chatButton.setText("Отправить");
        chatButton.addActionListener(e -> {
            Bot[] selectedBots = Main.getMainFrame().getSelectedBots();
            for (Bot bot : selectedBots) {
                bot.sendMessage(chatText.getText());
            }
        });
        commandPanel.getRootPane().setDefaultButton(chatButton);

        chatPanel.add(chatButton, new GridConstraints(1, 0, 1, 1, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_CAN_SHRINK | GridConstraints.SIZEPOLICY_CAN_GROW, GridConstraints.SIZEPOLICY_FIXED, null, null, null, 0, false));
        chatPanel.add(chatText, new GridConstraints(0, 0, 1, 1, GridConstraints.ANCHOR_WEST, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_WANT_GROW, GridConstraints.SIZEPOLICY_FIXED, null, new Dimension(150, -1), null, 0, false));

        commandPanel.add(chatPanel, new GridConstraints(1, 0, 1, 1, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_CAN_SHRINK | GridConstraints.SIZEPOLICY_CAN_GROW, GridConstraints.SIZEPOLICY_CAN_SHRINK | GridConstraints.SIZEPOLICY_CAN_GROW, null, null, null, 0, false));
    }
}

File: src/main/java/me/xjcyan1de/cyanbot/gui/commands/CommandChat.kt
package me.xjcyan1de.cyanbot.gui.commands

import com.intellij.uiDesigner.core.GridConstraints
import com.intellij.uiDesigner.core.GridLayoutManager
import me.xjcyan1de.cyanbot.Main

import javax.swing.*
import java.awt.*

class CommandChat : Command("Сообщение в чат") {

    override fun initPanel(commandPanel: JPanel) {
        val chatPanel = JPanel()
        chatPanel.layout = GridLayoutManager(2, 1, Insets(0, 0, 0, 0), -1, -1)
        chatPanel.border = BorderFactory.createTitledBorder("Чат")

        val chatText = JTextField()
        val chatButton = JButton()
        chatButton.text = "Отправить"
        chatButton.addActionListener { _ ->
            val selectedBots = Main.getMainFrame().selectedBots
            for (bot in selectedBots) {
                bot.sendMessage(chatText.text)
            }
        }
        commandPanel.rootPane.defaultButton = chatButton

        chatPanel.add(chatButton, GridConstraints(1, 0, 1, 1, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_CAN_SHRINK or GridConstraints.SIZEPOLICY_CAN_GROW, GridConstraints.SIZEPOLICY_FIXED, null, null, null, 0, false))
        chatPanel.add(chatText, GridConstraints(0, 0, 1, 1, GridConstraints.ANCHOR_WEST, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_WANT_GROW, GridConstraints.SIZEPOLICY_FIXED, null, Dimension(150, -1), null, 0, false))

        commandPanel.add(chatPanel, GridConstraints(1, 0, 1, 1, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_CAN_SHRINK or GridConstraints.SIZEPOLICY_CAN_GROW, GridConstraints.SIZEPOLICY_CAN_SHRINK or GridConstraints.SIZEPOLICY_CAN_GROW, null, null, null, 0, false))
    }
}
